galley/pkg/kube/source/init.go

```diff
@@ -15,18 +15,86 @@
 package source
 
 import (
+	"io/ioutil"
 	"fmt"
+	"os"
 	"time"
+	"strings"
 
+	"github.com/ghodss/yaml"
 	"github.com/hashicorp/go-multierror"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/wait"
-
 	"istio.io/istio/galley/pkg/kube"
 	kube_meta "istio.io/istio/galley/pkg/metadata/kube"
 )
 
+func writeCRDs(cs clientset.Interface, fileName string) (error) {
+	CRDData, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		scope.Infof("Read CRD %v", err)
+		return err
+	}
+
+	// yaml.YAMLtoJSON does not appear to understand "---".
+	yamlStrings := strings.Split(string(CRDData), "---")
```

Comment:
Consider using [`yaml.NewYAMLToJSONDecoder`](https://github.com/kubernetes/apimachinery/blob/4a9a8137c0a17bc4594f544987b3f0d48b2e3d3a/pkg/util/yaml/decoder.go#L55) here instead of writing a custom YAML document parser. 

Alternatively, you could also look into using `resource.Builder{}` to handle the file parsing also. See [here](https://github.com/istio/istio/blob/3a311b51fbb95d2449e68317518056fc07c3fd1c/istioctl/pkg/validate/validate.go#L99) for an example usage.